File: code/intermediate/inheritance/Inheritance.py
class Parent:
    def __init__(self, name):
        self.name = name
        print("Parent constructor called")

    def printName(self):
        return f"Name :: {self.name}"

    def __str__(self):
        return f"{self.name}"


class Child(Parent):
    def __init__(self, name, age):
        # Parent.__init__(self, name)
        super().__init__(name)
        self.age = age
        print("Child constructor called")

    def printAge(self):
        return f"Name :: {self.age}"

    def __str__(self):
        return f"{self.age}"


c = Child("Avinash", 21)
print(c.printAge())
print(c.printName())

File: code/intermediate/inheritance/Inheritance.java
package code.intermediate.inheritance;

class Parent {
    String name;

    Parent(String name) {
        this.name = name;
        System.out.println("Parent constructor called");
    }

    void printName() {
        System.out.println(this.name);
    }
}

class Child extends Parent {
    int age;

    Child(int age, String name) {
        super(name);
        this.age = age;
        System.out.println("Child constructor called");
    }

    void printAge() {
        System.out.println(this.age);
    }

    public static void main(String args[]) {
        Child c = new Child(21, "Avinash");
        c.printAge();
        c.printName();
    }
}